unit_testing/cli/test_cli_sct_detect_pmj.py

```diff
@@ -1,23 +1,40 @@
-from pytest_console_scripts import script_runner
 import pytest
 import logging
-import os
+
+import numpy as np
+
+from spinalcordtoolbox.image import Image
 from spinalcordtoolbox.scripts import sct_detect_pmj
 
 logger = logging.getLogger(__name__)
 
 
-@pytest.mark.script_launch_mode('subprocess')
-def test_sct_detect_pmj_backwards_compat(script_runner):
-    ret = script_runner.run('sct_testing', '--function', 'sct_detect_pmj')
-    logger.debug(f"{ret.stdout}")
-    logger.debug(f"{ret.stderr}")
-    assert ret.success
-    assert ret.stderr == ''
-
-def test_sct_detect_pmj_o_flag(tmp_path):
-    argv = ['-i', 'sct_testing_data/template/template//PAM50_small_t2.nii.gz', '-igt',
-            'sct_testing_data/template/template/PAM50_small_t2_pmj_manual.nii.gz',
-            '-c', 't2', '-o', os.path.join(str(tmp_path), 'test_pmj.nii.gz')]
-    sct_detect_pmj.main(argv)
-    assert os.path.isfile(os.path.join(str(tmp_path), 'test_pmj.nii.gz'))
+@pytest.mark.sct_testing
+@pytest.mark.usefixtures("run_in_sct_testing_data_dir")
+def test_sct_detect_pmj_check_euclidean_distance_against_groundtruth():
+    """Run the CLI script and verify that euclidean distances between predicted and ground truth coordinates
+    are within a certain threshold."""
+    fname_in = 'template/template/PAM50_small_t2.nii.gz'
+    fname_out = 'PAM50_small_t2_pmj.nii.gz'
+    fname_gt = 'template/template/PAM50_small_t2_pmj_manual.nii.gz'
+    sct_detect_pmj.main(argv=['-i', fname_in, '-o', fname_out, '-c', 't2', '-qc', 'testing-qc'])
+
+    im_pmj = Image(fname_out)
+    im_pmj_manual = Image(fname_gt)
+
+    # np.where outputs a tuple with an array for each axis: https://stackoverflow.com/q/50646102
+    # so, convert ==> `tuple(x_array, y_array, z_array)` -> `list([x1, y1, z1], [x2, y2, z2], ...)`
+    label_value = 50
+    gt = [[x, y, z] for x, y, z in zip(*np.where(im_pmj_manual.data == label_value))]
+    pred = [[x, y, z] for x, y, z in zip(*np.where(im_pmj.data == label_value))]
```

Comment:
I find this a bit easier to read. But I'm familiar with `zip`. But I think it's a pretty basic part of python vocabulary.

```suggestion
    gt = list(zip(*np.where(im_pmj_manual.data == label_value)))
    pred = list(zip(*np.where(im_pmj.data == label_value)))
```

Reply:
Oh! Of course, that makes perfect sense.

Reply:
Ah, wait, now I remember why I did it this way. I wanted to make it a little more explicit that the result was a list of `[x, y, z]` coordinates.

`list()` is definitely more compact, but maybe a little harder to grok (i.e. more dependent on the comment above to explain).